apps/babanews/pages/a/index.tsx

```diff
@@ -13,11 +25,38 @@ export default function ArticleList() {
     }),
     {
       sort: ApiV1.ArticleSort.PublishedAt,
-      order: ApiV1.SortOrder.Descending
+      order: ApiV1.SortOrder.Descending,
+      skip: 0,
+      take: limit
     } as Partial<ApiV1.ArticleListQueryVariables>
   )
 
-  return <ArticleListContainer variables={variables} onVariablesChange={onVariablesChange} />
+  const {data} = ApiV1.useArticleListQuery({
+    fetchPolicy: 'cache-only',
+    variables
+  })
+
+  const paginationCount = useMemo(() => {
+    if (data?.articles.totalCount && data?.articles.totalCount > limit) {
+      return Math.ceil(data.articles.totalCount / limit)
+    }
+    return 1
+  }, [data?.articles.totalCount, limit])
+
+  useEffect(() => {
```

Comment:
Not needed and causes a double render, call onVariablesChange inside the onChange

Reply:
Actually no, the way this was implemented we need the setPage. But we can move this also into the onChange to not have a double render.